Mixpanel/MPABTestDesignerConnection.m

```diff
@@ -255,21 +259,78 @@ - (void)webSocket:(MPWebSocket *)webSocket didCloseWithCode:(NSInteger)code reas
+        [self.connectivityIndicatorWindow addGestureRecognizer:panRecognizer];
+        indicatorAnimator = [[UIDynamicAnimator alloc] initWithReferenceView:self.connectivityIndicatorWindow];
+        
+        [UIView animateWithDuration:0.3 animations:^{
+            self.connectivityIndicatorWindow.alpha = 1;
+        }];
     }
 }
 
 - (void)hideConnectedView
 {
-    if (_recordingView) {
-        [_recordingView removeFromSuperview];
+    if (self.connectivityIndicatorWindow) {
+        [self.connectivityIndicatorWindow setHidden:YES];
+    }
+    self.connectivityIndicatorWindow = nil;
+}
+
+- (void)moveConnectivityIndicator:(UIPanGestureRecognizer *)gesture
+{
+    if (!self.connectivityIndicatorWindow) {
+        return;
+    }
+    
+    CGPoint currPoint = [gesture locationInView:gesture.view];
+    switch (gesture.state) {
+        case UIGestureRecognizerStateBegan: {
+            [indicatorAnimator removeAllBehaviors];
+            indicatorPrevPoint = currPoint;
+            break;
+        }
+        case UIGestureRecognizerStateEnded: {
+            UIWindow *mainWindow = [UIApplication sharedApplication].keyWindow;
+            CGPoint velocity = [gesture velocityInView:mainWindow];
+            CGFloat magnitude = sqrt((velocity.x * velocity.x) + (velocity.y * velocity.y));
+            
+            UICollisionBehavior* collisionBehavior = [[UICollisionBehavior alloc] initWithItems:@[self.connectivityIndicatorWindow]];
+            UIBezierPath *path = [UIBezierPath bezierPathWithRect:[UIScreen mainScreen].bounds];
+            [collisionBehavior addBoundaryWithIdentifier:@"border" forPath:path];
+            [indicatorAnimator addBehavior:collisionBehavior];
+            
+            UIPushBehavior *pushBehavior = [[UIPushBehavior alloc]
+                                            initWithItems:@[self.connectivityIndicatorWindow]
+                                            mode:UIPushBehaviorModeInstantaneous];
+            pushBehavior.pushDirection = CGVectorMake((velocity.x / 10) , (velocity.y / 10));
+            pushBehavior.magnitude = magnitude / 500;
+            [indicatorAnimator addBehavior:pushBehavior];
+            
+            UIDynamicItemBehavior *itemBehavior = [[UIDynamicItemBehavior alloc] initWithItems:@[self.connectivityIndicatorWindow]];
+            itemBehavior.resistance = 10;
+            itemBehavior.allowsRotation = NO;
+            [indicatorAnimator addBehavior:itemBehavior];
+            break;
+        }
+        case UIGestureRecognizerStateChanged: {
+            gesture.view.center = CGPointMake(MIN(MAX(gesture.view.center.x + (currPoint.x - indicatorPrevPoint.x),CONNECTIVITY_INDICATOR_SIZE/2),[UIScreen mainScreen].bounds.size.width - CONNECTIVITY_INDICATOR_SIZE/2), MIN(MAX(gesture.view.center.y + (currPoint.y - indicatorPrevPoint.y),CONNECTIVITY_INDICATOR_SIZE/2),[UIScreen mainScreen].bounds.size.height - CONNECTIVITY_INDICATOR_SIZE/2));
```

Comment:
This is way too long for one line, please add some intermediate variables so we can parse it easily.